<?xml version="1.0"?>
<launch>
	<!-- Select the robot -->
  <arg name="robot_name" default="iiwa" />
  <arg name="model" default="14" />

  <!-- Select the controller -->
  <arg name="controller" default="torque"/>

  <!-- Select the hardware interface (probably used in the urdf) -->
  <arg if="$(eval controller == 'torque')" name="hardware_interface" value="hardware_interface/EffortJointInterface"/>
  <arg if="$(eval controller == 'pid')" name="hardware_interface" value="hardware_interface/EffortJointInterface"/>
  <arg if="$(eval controller == 'position')" name="hardware_interface" value="hardware_interface/PositionJointInterface"/>

  <!-- Upload iiwa URDF -->
  <include file="$(find iiwa_description)/launch/$(arg robot_name)$(arg model)_upload.launch">
    <arg name="hardware_interface" value="$(arg hardware_interface)"/>
    <arg name="robot_name" value="$(arg robot_name)"/>
  </include>

  <!-- <rosparam file="$(find iiwa_hardware_interface)/config/joint_limits.yaml" command="load"/> -->
  
  <!-- Load iiwa hardware interface -->
	<node pkg="iiwa_hw" type="iiwa_hw_node" name="iiwa_hw" respawn="false" output="screen" >
    <remap from="/joint_states" to="/iiwa/joint_states"/>
    <remap from="/controller_manager" to="/iiwa/controller_manager"/>
    <!-- Load configurations from YAML file to parameter server -->
    <rosparam file="$(find iiwa_hw)/config/hardware.yaml" command="load"/>
  </node>

  <!-- Load controller -->
  <include file="$(find iiwa_control)/launch/iiwa_control.launch">
    <arg name="robot_name" value="$(arg robot_name)" />
    <arg name="model" value="$(arg model)" />
    <arg name="controller" value="$(arg controller)"/>
  </include>
  
</launch>